packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1385,6 +1385,20 @@ class OffsetLayer extends ContainerLayer {
     properties.add(DiagnosticsProperty<Offset>('offset', offset));
   }
 
+  ui.Scene _createSceneForImage(Rect bounds, { double pixelRatio = 1.0 }) {
+    assert(bounds != null);
+    assert(pixelRatio != null);
+    final ui.SceneBuilder builder = ui.SceneBuilder();
+    final Matrix4 transform = Matrix4.translationValues(
+      (-bounds.left  - offset.dx) * pixelRatio,
+      (-bounds.top - offset.dy) * pixelRatio,
+      0.0,
+    );
+    transform.scale(pixelRatio, pixelRatio);
```

Comment:
It is a poorly named method in Matrix4, but I think it would be more straightforward to just say:

```
  final Matrix4 transform = Matrix4.diagonal3Values(pixelRatio, pixelRatio, 1);
  transform.translate(-(bounds.left + offset.dx), -(bounds.top + offset.dy));
```

Reply:
I wonder whether the two are equivalent? The original code is "first translate then scale" while the suggested code is "first scale then translate"